pkg/hintrunner/zero/zerohint_ec.go

```diff
@@ -0,0 +1,72 @@
+package zero
+
+import (
+	"fmt"
+	"github.com/NethermindEth/cairo-vm-go/pkg/hintrunner/hinter"
+	secp_utils "github.com/NethermindEth/cairo-vm-go/pkg/hintrunner/utils"
+	VM "github.com/NethermindEth/cairo-vm-go/pkg/vm"
+	"github.com/consensys/gnark-crypto/ecc/stark-curve/fp"
+	"math/big"
+)
+
+func newEcNegateHint(point hinter.ResOperander) hinter.Hinter {
+	return &GenericZeroHinter{
+		Name: "EcNegate",
+		Op: func(vm *VM.VirtualMachine, ctx *hinter.HintRunnerContext) error {
+			//> from starkware.cairo.common.cairo_secp.secp_utils import SECP_P, pack
+			//> y = pack(ids.point.y, PRIME) % SECP_P
+			//> # The modulo operation in python always returns a nonnegative number.
+			//> value = (-y) % SECP_P
+
+			secPBig, ok := secp_utils.GetSecPBig()
+			if !ok {
+				return fmt.Errorf("GetSecPBig failed")
+			}
+
+			pointValues, err := hinter.GetConsecutiveValues(vm, point, int16(6))
+			if err != nil {
+				return err
+			}
+
+			// [y.d0, y.d1, y.d2]
+			var pointValuesBig [3]*big.Int
+			for i := 0; i < 3; i++ {
+				pointValue, err := pointValues[i+3].FieldElement()
+				if err != nil {
+					return err
+				}
+				pointValueBig := pointValue.BigInt(new(big.Int))
+				pointValuesBig[i] = pointValueBig
+			}
+
+			primeBig := fp.Modulus()
+
+			//> y = pack(ids.point.y, PRIME) % SECP_P
+			yBig, err := secp_utils.SecPPacked(pointValuesBig[0], pointValuesBig[1], pointValuesBig[2], primeBig)
```

Comment:
Could we consider passing `pointValuesBig` instead of each value separately? The `pack` method, as far as I've seen, always receives a triple as argument and you have to actually iterate through its members in the implementation

Reply:
Yeah I actually had the same thought before. I can modify the method to receive a slice of size 3. What do you think of defining a custom type for this though? A type with d0, d1 and d2 fields? Overkill?

Reply:
Just used a `[3]*fp.Element` with a comment.

Reply:
Yeah, I thought about a custom type for this as well, but not sure if it will be worth it. Do we have to use this structure very often? Besides the `pack` method, do we need to implement any other functionality for it? If that's not the case, then I would stick to the current solution with the slice